localboot/grub.go

```diff
@@ -85,6 +87,31 @@ func ParseGrubCfg(grubcfg string, basedir string, grubVersion int) []bootconfig.
 			} else if sline[0] == "initrd" || sline[0] == "initrd16" || sline[0] == "initrdefi" {
 				initrd := sline[1]
 				cfg.Initramfs = path.Join(basedir, initrd)
+			} else if sline[0] == "multiboot" {
+				multiboot := sline[1]
+				cmdline := strings.Join(sline[2:], " ")
+				if grubVersion == 2 {
```

Comment:
this block is duplicated from above. Can you please refactor into its own function?
Something like:

```
type grubVersion int

var (
	grubV1 grubVersion = 1
	grubV2 grubVersion = 2
)

func unquoteGrub(ver grubVersion, text string) {
	if ver == grubV2 {
		// if grub2, unquote the string, as directives could be quoted
		// https://www.gnu.org/software/grub/manual/grub/grub.html#Quoting
		// TODO unquote everything, not just \$
		return strings.Replace(text, `\$`, "$", -1)
	}
	return text
}
```

then use it like this:
```
cmdline = unquoteGrub(ver, cmdline)
```

Note that here I introduced a new type `grubVersion`, that has the same name a the variable passed as input of `ParseGrubCfg`. You should rename such variable into something different, e.g. just `ver`.

Reply:
I committed this change on your branch as I mentioned to @zaolin offline, feel free to revert it if you prefer to do it differently